src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsCodeFixProvider.cs

```diff
@@ -0,0 +1,218 @@
+      IfAndElseIfStatementSyntaxManipulator syntaxManipulator,
+      List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes)
+    {
+      if (syntaxManipulator.ElseClauseHasIfStatement())
+      {
+        var heldIfStatement = syntaxManipulator.ElseClause!.Statement as IfStatementSyntax;
+        var elseClauseSyntaxWrapper = CreateSyntaxWrapper(heldIfStatement!);
+        FillSyntaxNodeListWithStatements(syntaxNodes, elseClauseSyntaxWrapper, heldIfStatement!.Condition);
+      }
+      else
+      {
+        syntaxNodes.Add(new IfAndElseIfStatementSyntaxManipulator(syntaxManipulator.ElseClause!, null));
+      }
+
+      ReintroduceParentElseClauses(syntaxNodes);
+    }
+
+    private static IfAndElseIfStatementSyntaxManipulator CreateSyntaxWrapper (IfStatementSyntax ifStatement)
+    {
+      IfAndElseIfStatementSyntaxManipulator blueprint;
+      if (ifStatement.Parent.IsKind(SyntaxKind.ElseClause))
+        blueprint = new IfAndElseIfStatementSyntaxManipulator(ifStatement.Parent, ifStatement.Else);
+      else
+        blueprint = new IfAndElseIfStatementSyntaxManipulator(ifStatement, ifStatement.Else);
+
+      return blueprint;
+    }
+
+    private static SyntaxNode BuildRoot (
+      SyntaxNode oldRoot,
+      List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes,
+      IfAndElseIfStatementSyntaxManipulator oldNode)
+    {
+      BuildProperStatementList(syntaxNodes);
+
+      //ReplaceNode() can only replace an ifNode by a list if the list does not include else statements.
+      //Else if statements can only be in the list if it's count is 1
+      if (syntaxNodes.Count == 1)
+        return oldRoot.ReplaceNode(oldNode.IfOrElseIfStatement, syntaxNodes.First().IfOrElseIfStatement);
+
+      return oldRoot.ReplaceNode(
+        oldNode.IfOrElseIfStatement, 
+        syntaxNodes.Select(s => s.IfOrElseIfStatement));
+    }
+
+    private static void BuildProperStatementList (List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes)
+    {
+      var tempElseClause = syntaxNodes.Last();
+      syntaxNodes.RemoveAt(syntaxNodes.Count - 1);
+
+      while (syntaxNodes.Count > 0)
+      {
+        if (tempElseClause.IfOrElseIfStatement.IsKind(SyntaxKind.IfStatement))
+          break;
+
+        var tempParentElseClause = syntaxNodes.Last();
+
+        tempParentElseClause.WithElse((ElseClauseSyntax)tempElseClause.IfOrElseIfStatement);
+        tempElseClause = tempParentElseClause;
+        syntaxNodes.RemoveAt(syntaxNodes.Count - 1);
+      }
+
+      syntaxNodes.Add(tempElseClause);
+    }
+
+
+    private static void FillSyntaxNodeListWithStatements (
+      List<IfAndElseIfStatementSyntaxManipulator> syntaxNodes,
+      IfAndElseIfStatementSyntaxManipulator blueprint,
+      SyntaxNode expression)
+    {
+      if (SimpleIfStatementDetectorUtility.IsIllegalExpression(expression, out var illegalExpression))
+      {
+        var splitExpression = SplitExpressionUtility.SplitExpression(illegalExpression!);
```

Comment:
there's NRT annotations for this to be applied on IsIllegalExpression. you don't need to rely on !.
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute?view=net-6.0

Reply:
I think we arent using a version of dotnet that supports these attributes yet, we are using .net standard 2.0, our roslyn assertion library only works with it and framework 4.7.2 I believe.